go/vt/worker/legacy_split_clone.go

```diff
@@ -609,85 +579,47 @@ func (scw *LegacySplitCloneWorker) copy(ctx context.Context) error {
 		return firstError
 	}
 
-	// then create and populate the blp_checkpoint table
-	if scw.strategy.skipPopulateBlpCheckpoint {
-		scw.wr.Logger().Infof("Skipping populating the blp_checkpoint table")
-	} else {
-		queries := make([]string, 0, 4)
-		queries = append(queries, binlogplayer.CreateBlpCheckpoint()...)
-		flags := ""
-		if scw.strategy.dontStartBinlogPlayer {
-			flags = binlogplayer.BlpFlagDontStart
-		}
-
-		// get the current position from the sources
-		for shardIndex := range scw.sourceShards {
-			shortCtx, cancel := context.WithTimeout(ctx, *remoteActionsTimeout)
-			status, err := scw.wr.TabletManagerClient().SlaveStatus(shortCtx, scw.sourceTablets[shardIndex])
-			cancel()
-			if err != nil {
-				return err
-			}
-
-			queries = append(queries, binlogplayer.PopulateBlpCheckpoint(uint32(shardIndex), status.Position, scw.maxTPS, throttler.ReplicationLagModuleDisabled, time.Now().Unix(), flags))
-		}
-
-		for _, si := range scw.destinationShards {
-			destinationWaitGroup.Add(1)
-			go func(keyspace, shard string) {
-				defer destinationWaitGroup.Done()
-				scw.wr.Logger().Infof("Making and populating blp_checkpoint table")
-				keyspaceAndShard := topoproto.KeyspaceShardString(keyspace, shard)
-				if err := runSQLCommands(ctx, scw.wr, scw.tsc, keyspace, shard, scw.destinationDbNames[keyspaceAndShard], queries); err != nil {
-					processError("blp_checkpoint queries failed: %v", err)
-				}
-			}(si.Keyspace(), si.ShardName())
-		}
-		destinationWaitGroup.Wait()
-		if firstError != nil {
-			return firstError
-		}
-	}
-
-	// Now we're done with data copy, update the shard's source info.
-	// TODO(alainjobart) this is a superset, some shards may not
-	// overlap, have to deal with this better (for N -> M splits
-	// where both N>1 and M>1)
-	if scw.strategy.skipSetSourceShards {
-		scw.wr.Logger().Infof("Skipping setting SourceShard on destination shards.")
-	} else {
-		for _, si := range scw.destinationShards {
-			scw.wr.Logger().Infof("Setting SourceShard on shard %v/%v", si.Keyspace(), si.ShardName())
-			shortCtx, cancel := context.WithTimeout(ctx, *remoteActionsTimeout)
-			err := scw.wr.SetSourceShards(shortCtx, si.Keyspace(), si.ShardName(), scw.sourceAliases, nil)
-			cancel()
-			if err != nil {
-				return fmt.Errorf("failed to set source shards: %v", err)
-			}
+	sourcePositions := make([]string, len(scw.sourceShards))
+	// get the current position from the sources
+	for shardIndex := range scw.sourceShards {
+		shortCtx, cancel := context.WithTimeout(ctx, *remoteActionsTimeout)
+		status, err := scw.wr.TabletManagerClient().SlaveStatus(shortCtx, scw.sourceTablets[shardIndex])
+		cancel()
+		if err != nil {
+			return err
 		}
+		sourcePositions[shardIndex] = status.Position
 	}
 
-	err = scw.findRefreshTargets(ctx)
```

Comment:
Correct. It was necessary because the uid is now returned by the vreplication `create` command. So, using it right away to create the source shard makes sense. The UID was previously inferred by the order of the shards at a different location. That approach felt too flimsy, and didn't offer any particular advantages.